projects/igniteui-angular/src/lib/grid/api.service.ts

```diff
@@ -388,17 +388,22 @@ export class IgxGridAPIService {
     public groupBy_toggle_group(id: string, groupRow: IGroupByRecord) {
         const grid = this.get(id);
         const expansionState = grid.groupingExpansionState;
+        let expanded: boolean;
 
         const state: IGroupByExpandState = this.groupBy_get_expanded_for_group(id, groupRow);
         if (state) {
-            state.expanded = !state.expanded;
+            state.expanded = expanded = !state.expanded;
         } else {
+            expanded = !grid.groupsExpanded;
             expansionState.push({
-                expanded: !grid.groupsExpanded,
+                expanded,
                 hierarchy: DataUtil.getHierarchy(groupRow)
             });
         }
         this.get(id).groupingExpansionState = expansionState;
+        if (grid.rowEditable) {
+            grid.toggleRowEditingOverlay(expanded);
```

Comment:
@Lipata Wouldn't the fix also show a row editing overlay after expanding a group if there is no row in edit mode?

Reply:
Yes, it will show the overlay when group is expanded, because the cell(row) is still in edit mode (this is also valid, when row editing is disabled). Basically the overlay is just hidden (like it is when scrolling grid and overlay goes outside visible area, but it is still in edit mode). The other thing is that collapsing and expanding is not data changing operation, so in that case we are not forced to finish editing operation and submit changes.

Reply:
No, it will not show overlay, because showRowEditingOverlay is just dealing with display style property of already opened overlay (the latter happens, when overlay.open() method is used). We use open and close methods of the overlay, only when we finish row editing, because they trigger detect changes. For cases, when cell(row) is still in edit mode, we only show/hide overlay.

Reply:
The overlay display will still be correct if after collapse a different row enters edit mode, right?

Reply:
There are a couple of issues related to this.
1.	The overlay is not repositioned correctly. When edit a row and collapse the group, edit in another group’s row and expand the previous/collapsed one, the overlay (row editing buttons) is still there and not next to the currently edited row.
 
![image](https://user-images.githubusercontent.com/4296944/47500318-2b0a5d80-d86b-11e8-97b4-b3dfbd7f3065.png)

2.	The overlay is not repositioned correctly. When edit a row and collapse the group, then collapse and expand a different one, the overlay (row editing buttons) position is wrong.

![image](https://user-images.githubusercontent.com/4296944/47500326-3067a800-d86b-11e8-8e8b-a36197aaa8b6.png)

Reply:
In addition to those scenarios, I've also added test for hierarchical groups.